assembl/models/langstrings.py

```diff
@@ -267,14 +284,31 @@ def get_or_create(cls, locale_code, db=None):
             locale_object_cache[locale_code] = locale
             return locale
         # create it.
-        locale = Locale(code=locale_code)
+        locale = Locale(
+            code=locale_code, rtl=cls.is_right_to_left_code(locale_code))
         db.add(locale)
         db.flush()
         cls._locale_uncommitted.append(locale)
         cls.reset_cache()
         locale_object_cache[locale_code] = locale
         return locale
 
+    @classmethod
+    def bulk_get_or_create(cls, locale_codes, db=None):
+        db = db or cls.default_db
+        db.execute("lock table %s in exclusive mode" % cls.__table__.name)
```

Comment:
Not sure if this is accurate, but should the share mode be added to this statement for greater control over the concurrent change of locales? https://www.postgresql.org/docs/9.4/static/sql-lock.html

Reply:
you mean share instead of exclusive? (It would replace, not add. There is no share exclusive.)
Doc is weird; it says share precludes concurrent data access, yet it does not conflict with the row share option, i.e. another update. I think exclusive is safer, and not that much harder to obtain (unless alembic is running, which would be weird.)